sims/xcelium/Makefile

```diff
@@ -0,0 +1,138 @@
+# xcelium build paths
+#########################################################################################
+model_dir = $(build_dir)/$(long_name)
+model_dir_debug = $(build_dir)/$(long_name).debug
+
+
+#########################################################################################
+# xcelium simulator rules
+#########################################################################################
+
+$(sim_workdir): $(sim_common_files) $(dramsim_lib) $(EXTRA_SIM_REQS)
+	rm -rf $(model_dir)
+	$(XCELIUM) -elaborate $(XCELIUM_OPTS) $(EXTRA_SIM_SOURCES) $(XCELIUM_COMMON_ARGS)
+
+$(sim_run_tcl): $(sim_workdir)
+	echo "$$CAD_INFO_HEADER" > $(sim_run_tcl)
+	echo "run" >> $(sim_run_tcl)
+	echo "exit" >> $(sim_run_tcl)
+
+$(sim): $(sim_workdir) $(sim_run_tcl)
+	echo "#!/usr/bin/env bash" > $(sim)
+	echo "$$CAD_INFO_HEADER" >> $(sim)
+	cat arg-reshuffle >> $(sim)
+	echo "LD_PRELOAD=$(base_dir)/.conda-env/lib/libstdc++.so.6 $(XCELIUM) +permissive -R -input $(sim_run_tcl) $(XCELIUM_COMMON_ARGS) +permissive-off \$$INPUT_ARGS" >> $(sim)
+	chmod +x $(sim)
+
+$(sim_debug_run_tcl): $(sim_workdir)
+	echo "$$CAD_INFO_HEADER" > $(sim_debug_run_tcl)
+	echo "database -open default_vcd_dump -vcd -into \$$env(XCELIUM_WAVEFORM_FLAG)" >> $(sim_debug_run_tcl)
+	echo "set probe_packed_limit 64k" >> $(sim_debug_run_tcl)
+	echo "probe -create $(TB) -database default_vcd_dump -depth all -all" >> $(sim_debug_run_tcl)
+	echo "run" >> $(sim_debug_run_tcl)
+	echo "database -close default_vcd_dump" >> $(sim_debug_run_tcl)
+	echo "exit" >> $(sim_debug_run_tcl)
+
+
+$(sim_debug): $(sim_workdir) $(sim_debug_run_tcl)
+	echo "#!/usr/bin/env bash" > $(sim_debug)
+	echo "$$CAD_INFO_HEADER" >> $(sim_debug)
+	cat arg-reshuffle >> $(sim_debug)
+	echo "export XCELIUM_WAVEFORM_FLAG=\$$XCELIUM_WAVEFORM_FLAG" >> $(sim_debug)
+	echo "LD_PRELOAD=$(base_dir)/.conda-env/lib/libstdc++.so.6 $(XCELIUM) +permissive -R -input $(sim_debug_run_tcl) $(XCELIUM_COMMON_ARGS) +permissive-off \$$INPUT_ARGS" >> $(sim_debug)
+	chmod +x $(sim_debug)
+
+
+#########################################################################################
+# create vcd rules
+#########################################################################################
+.PRECIOUS: $(output_dir)/%.vcd %.vcd
+$(output_dir)/%.vcd: $(output_dir)/% $(sim_debug)
+	(set -o pipefail && $(sim_debug) $(PERMISSIVE_ON) $(SIM_FLAGS) $(EXTRA_SIM_FLAGS) $(SEED_FLAG) $(VERBOSE_FLAGS) +vcdplusfile=$@ $(PERMISSIVE_OFF) $< </dev/null 2> >(spike-dasm > $<.out) | tee $<.log)
+
+#########################################################################################
+# general cleanup rules
+#########################################################################################
+.PHONY: clean clean-sim clean-sim-debug
+clean:
+	rm -rf $(gen_dir) $(sim_prefix)-* ucli.key
+
+clean-sim:
+	rm -rf $(model_dir) $(build_dir)/vc_hdrs.h $(sim) $(sim_workdir) $(sim_run_tcl) ucli.key bpad_*.err sigusrdump.out dramsim*.log
```

Comment:
Ditto: Is the vc_hdrs.h needed?

Reply:
Does the `.gitignore` need to be updated for these files? `.gitignore` in this new folder?